In this assignment I will empirically verify the concept of confidence intervals. A confidence interval is a range of values in which we are confident the true value of a parameter lies in.

Specifically, I will be using a Normal distribution with mean $\mu = 3$ and variance $\sigma^2 = 25$, i.e. $N(3,25)$.

The theoretical $(1-\alpha)$% confidence interval for a $N(\mu,\sigma^2)$ distribution for the parameter $\mu$ is $\mu \pm z_{\alpha/2}\frac{\sigma}{\sqrt{n}}$ where n is the number of observations in the sample.

I will generate 100 random samples of 1000 observations (n=1000) in order to see how many times the confidence bound is breached compared to 95%, 90%, and 80% confidence.

First we calculate the theoretical 95% confidence interval, so $\alpha = 0.05$.
From a Normal Curve Area table we get that $z_{0.025} = 1.96$.

In [419]:
l_bound = 3 - (1.96*(5/1000**0.5))
u_bound = 3 + (1.96*(5/1000**0.5))
print("Lower Bound:",round(l_bound,2))
print("Upper Bound:",round(u_bound,2))

Lower Bound: 2.69
Upper Bound: 3.31


Thus, the 95% confidence interval for the mean $\mu$ of a N(3,25) with 1000 observations is [2.69 , 3.31].

In [420]:
import numpy as np
n= 1000
s = 100
# creating an array with 100 samples with 1000 observations each of a normal distribution
# with mean 3 and standard deviation 5
np.random.seed(834920347)
samples = np.random.normal(3,5,size =(s,n))
print("shape of samples:", sample.shape)

shape of samples: (100, 1000)


In [421]:
samples

array([[ -6.58049047,   2.82135723,   9.63384165, ...,   4.04994403,
          9.10682189,   2.26962127],
       [ 19.38101175,  -1.14068869,   0.83794923, ...,   3.13569595,
         -1.44113795,   1.18646105],
       [  0.74896961,  -5.94013537,   8.24467059, ...,  -2.33065537,
          3.72962942,   1.85064374],
       ..., 
       [ -2.20456576,   2.65097802,  14.96749523, ...,   7.97051682,
          6.49084567,  -2.24634139],
       [-12.01919328,   8.14415992,   8.84928235, ...,  -3.91341474,
          1.59928907,  -7.61376978],
       [  1.46035428,   3.75244024,  13.48327691, ...,  -6.83932634,
          4.24886132,   7.3017669 ]])

In [422]:
# find mean of each sample
mean = []
for i in range(0,100):
    mean += [samples[i].mean()]

In [423]:
len(mean)

100

In [424]:
def count_breach_CI(mean = 1):
    count = 0
    for i in range(0,len(mean)):
        if mean[i] < l_bound or mean[i] > u_bound:
            count += 1
    return count

In [425]:
count95 = count_breach_CI(mean)
count95

4

Next we calculate the theoretical 90% confidence interval, so $\alpha = 0.1$.
From a Normal Curve Area table we get that $z_{0.05} = 1.645$.

In [426]:
l_bound = 3 - (1.645*(5/1000**0.5))
u_bound = 3 + (1.645*(5/1000**0.5))
print("Lower Bound:",round(l_bound,2))
print("Upper Bound:",round(u_bound,2))

Lower Bound: 2.74
Upper Bound: 3.26


In [427]:
count90 = count_breach_CI(mean)
count90

10

Last we calculate the theoretical 80% confidence interval, so $\alpha = 0.2$.
From a Normal Curve Area table we get that $z_{0.1} = 1.282$.

In [428]:
l_bound = 3 - (1.282*(5/1000**0.5))
u_bound = 3 + (1.282*(5/1000**0.5))
print("Lower Bound:",round(l_bound,2))
print("Upper Bound:",round(u_bound,2))

Lower Bound: 2.8
Upper Bound: 3.2


In [429]:
count80 = count_breach_CI(mean)
count80

20

We see that only 5 out of the 100 sample means lay outside of the 95% confidence interval [2.69 , 3.31], which means that 95/100 sample means were inside the confidence interval. In this case, we get exactly 95% of the sample means within the confidence interval as expected.